core/src/main/java/com/digitalpebble/stormcrawler/bolt/JSoupParserBolt.java

```diff
@@ -339,7 +345,10 @@ public void execute(Tuple tuple) {
         }
 
         if (emitOutlinks) {
-            for (Outlink outlink : parse.getOutlinks()) {
+            final List<Outlink> outlinksAfterLimit = (maxOutlinksPerPage == -1) ? parse
```

Comment:
I think i understand what is happening here. When i first looked at this i read "after limit" as the outlinks that exceeded the limit, not as the outlinks we are going to process.